File: src/main/scala/com/ScalaLR/scala/myNode.scala
package com.ScalaLR.scala
import java.io.Serializable

class myNode[E <: Comparable[E]](val data:E) extends Serializable{
  var next: myNode[E] = _ // Ссылка на следующий узел

  override def toString: String = data.toString// преобразование к строковому типу

  // Конструктор для инициализации
  def this(data: E, next: myNode[E]) = {
    this(data)
    this.next = next
  }
}

File: src/main/Java/com/JavaLr/java/myNode.java
package com.JavaLr.java;

import java.io.Serializable;

public class myNode<E extends Comparable<E>> implements Serializable {
    E data;//Экземпляр для данный
    myNode<E> next; // Сыылка на следующий узел
    //конструктор для иницализции
    myNode(E data){
        this.data = data;
        this.next = null;
    }
}
